uwblueprint/planet-read
# Fix: use only one add new language button

## Notion ticket link

https://www.notion.so/uwblueprintexecs/a658c933d18c48d4b43fd99bda2bd3b4?v=ba22ac93d09040ff8219c60ac7caabdb&p=25ebc25cf2a34ed9ae7efc19b8dd5a7b

## Implementation description

- removed add new language button outside of the approved languages table
- made add new language button in the table multi use for user vs admin
- renamed files:
  - ApprovedLanguageModal -> AdminNewLanguageModal
  - NewLanguageModal -> UserNewLanguageModal
- reoganized ApprovedLanguageTable to be more redable
- organized files: some files where scattered in utils, pages, and admin, when they shouldve been in userProfile

before: 
![image](https://user-images.githubusercontent.com/15113249/147700626-aeb744aa-0256-4e49-99c8-0f4fbccb865a.png)

after: 
![image](https://user-images.githubusercontent.com/15113249/147700604-48cd3a9f-9bb6-4a39-bfc1-24ddf0bbbc59.png)




## Steps to test

1. log in as carl
2. observe no out-of-table button
3. test adding language and level up
4. login as angela
5. observe no out-of-table button
6. test adding language, demoting, deleting languages



## What should reviewers focus on?

- does it work
- are the files nice n organized
- read ApprovedLanguagesTable. does the component as a whole makes sense? I had to rename some modal functions

## Checklist
